db/dcrpg/pgblockchain_test.go

```diff
@@ -50,29 +54,63 @@ func (m MemStats) String() string {
 
 var (
 	db           *ChainDB
+	sqlDb        *sql.DB
 	trefUNIX     int64 = 1454954400 // mainnet genesis block time
+	trefStr            = "2016-02-08T12:00:00-06:00"
 	addrCacheCap int   = 1e4
 )
 
+type dummyParser struct{}
+
+func (p *dummyParser) UpdateSignal() <-chan struct{} {
+	return make(chan struct{})
+}
+
+func (p *dummyParser) ProposalsHistory() ([]*pitypes.History, error) {
+	return []*pitypes.History{}, nil
+}
+
+func (p *dummyParser) ProposalsHistorySince(since time.Time) ([]*pitypes.History, error) {
+	return []*pitypes.History{}, nil
+}
+
 func openDB() (func() error, error) {
 	dbi := DBInfo{
-		Host:   "localhost",
-		Port:   "5432",
-		User:   "dcrdata", // postgres for admin operations
-		Pass:   "",
-		DBName: "dcrdata_mainnet_test",
+		Host:   dbconfig.PGTestsHost,
+		Port:   dbconfig.PGTestsPort,
+		User:   dbconfig.PGTestsUser,
+		Pass:   dbconfig.PGTestsPass,
+		DBName: dbconfig.PGTestsDBName,
 	}
 	var err error
 	db, err = NewChainDB(&dbi, &chaincfg.MainNetParams, nil, true, true,
-		addrCacheCap, nil, nil, nil)
+		addrCacheCap, nil, new(dummyParser), nil)
 	cleanUp := func() error { return nil }
 	if db != nil {
 		cleanUp = db.Close
+	} else {
+		return cleanUp, err
 	}
+
+	sqlDb = db.db
+	if err = DropTestingTable(sqlDb); err != nil {
+		return nil, err
+	}
+	if err = CreateTable(sqlDb, "testing"); err != nil {
+		return nil, err
+	}
+
 	return cleanUp, err
 }
 
 func TestMain(m *testing.M) {
+	// Setup dcrpg logger.
+	UseLogger(slog.NewBackend(os.Stdout).Logger("dcrpg"))
+	log.SetLevel(slog.LevelTrace)
+
+	// Setup db/cache logger.
+	cache.UseLogger(slog.NewBackend(os.Stdout).Logger("db/cache"))
```

Comment:
There was no logging from either dcrpg or cache, so we were missing the warnings about failed validation, which did not result in an error.  @buck54321 I'm wondering if we should have error values back from `Update`, etc.